api/src/app/modules/project/project.service.spec.ts

```diff
@@ -192,7 +192,7 @@ describe('ProjectService', () => {
         });
       
         afterEach(() => {
-            jest.resetAllMocks(); // very important if using mock/spy
+            jest.restoreAllMocks(); // very important if using mock/spy
```

Comment:
I need to update the `resetAllMocks` to `restoreAllMocks` to let the test pass. However I feel we need some revised on this test file to understand better what our spy `jest.spyOn(DateTimeUtil, 'isPNPostdateOnOrBeforeCommentingOpenDate')` is actually doing. 


By reading the latest documentation,  `jest.resetAllMocks()` is equivalent to calling `.mockReset()`, which will return **undefined**, and fail the test. `restoreAllMocks` will use the original non-mocked implementation, so I'm a little bit confused here then what we spy for. 
<img width="749" alt="Screen Shot 2024-01-03 at 4 52 06 PM" src="https://github.com/bcgov/nr-fom/assets/23131735/acccd311-1866-4a32-98c9-d9dffdca3ed4">
<img width="1116" alt="Screen Shot 2024-01-03 at 4 38 47 PM" src="https://github.com/bcgov/nr-fom/assets/23131735/9517e5ea-d8f3-4469-86d7-05a4e26e3b3e">

Their release note is not very clear when this change happened, the only thing I got is on [v29.6.3](https://github.com/jestjs/jest/releases/tag/v29.6.3) they reverted some breaking changes related to the jest-mock.

Reply:
Thank you Catherine for digging and finding out the breaking changes Jest had between versions!!!